source/particle/output/ascii.cc

```diff
@@ -106,19 +99,15 @@ namespace aspect
             output << "\n";
           }
 
-        file_index++;
+        this->increment_file_index ();
 
-        return output_file_prefix;
+        return "txt";
       }
 
       template <int dim>
       template <class Archive>
-      void ASCIIOutput<dim>::serialize (Archive &ar, const unsigned int)
-      {
-        // invoke serialization of the base class
-        ar &file_index
-        ;
-      }
+      void ASCIIOutput<dim>::serialize (Archive &, const unsigned int)
+      {}
```

Comment:
You can't make template functions virtual. You can only declare a function of the same name in a derived class, i.e., have the same template function in the derived class as in the base class. The function in the derived class would then serialize its own variables and then call the corresponding function in the base class that would serialize the variables in the base class (or do it the other way around).